include/dlaf/eigensolver/tridiag_solver/mc.h

```diff
@@ -0,0 +1,241 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <algorithm>
+
+#include <pika/future.hpp>
+#include <pika/unwrap.hpp>
+
+#include "dlaf/common/callable_object.h"
+#include "dlaf/eigensolver/tridiag_solver/api.h"
+#include "dlaf/eigensolver/tridiag_solver/merge.h"
+#include "dlaf/lapack/tile.h"
+#include "dlaf/matrix/copy_tile.h"
+#include "dlaf/sender/make_sender_algorithm_overloads.h"
+#include "dlaf/sender/policy.h"
+#include "dlaf/sender/transform.h"
+#include "dlaf/types.h"
+
+namespace dlaf::eigensolver::internal {
+
+template <class T>
+struct TridiagSolver<Backend::MC, Device::CPU, T> {
+  static void call(Matrix<T, Device::CPU>& mat_trd, Matrix<T, Device::CPU>& d,
+                   Matrix<T, Device::CPU>& mat_ev);
+};
+
+/// Splits [i_begin, i_end] in the middle and waits for all splits on [i_begin, i_middle] and [i_middle +
+/// 1, i_end] before saving the triad <i_begin, i_middle, i_end> into `indices`.
+///
+/// The recursive calls span a binary tree which is traversed in depth first left-right-root order. That
+/// is also the order of triads in `indices`.
+///
+/// Note: the intervals are all closed!
+///
+inline void splitIntervalInTheMiddleRecursively(
+    SizeType i_begin, SizeType i_end, std::vector<std::tuple<SizeType, SizeType, SizeType>>& indices) {
+  if (i_begin == i_end)
+    return;
+  SizeType i_middle = (i_begin + i_end) / 2;
+  splitIntervalInTheMiddleRecursively(i_begin, i_middle, indices);
+  splitIntervalInTheMiddleRecursively(i_middle + 1, i_end, indices);
+  indices.emplace_back(i_begin, i_middle, i_end);
+}
+
+/// Generates an array of triad indices. Each triad is composed of begin <= middle < end indices and
+/// represents two intervals [begin, middle] and [middle + 1, end]. The two intervals are the subprobelms
+/// that have to be merged to arrive at [begin, end].
+///
+/// Note: the intervals are all closed!
+///
+inline std::vector<std::tuple<SizeType, SizeType, SizeType>> generateSubproblemIndices(SizeType n) {
+  DLAF_ASSERT(n > 0, n);
+  std::vector<std::tuple<SizeType, SizeType, SizeType>> indices;
+  indices.reserve(to_sizet(n));
+  splitIntervalInTheMiddleRecursively(0, n - 1, indices);
+  return indices;
+}
+
+// Cuppen's decompostion
+//
+// Substracts the offdiagonal element at the split from the top and bottom diagonal elements and returns
+// the offdiagonal element. The split is between the last row of the top tile and the first row of the
+// bottom tile.
+//
+template <class T>
+T cuppensTileDecomposition(const matrix::Tile<T, Device::CPU>& top,
+                           const matrix::Tile<T, Device::CPU>& bottom) {
+  (void) top;
+  (void) bottom;
```

Comment:
```suggestion
  silenceUnusedWarningFor(top, bottom);
```
or... why is the `(void)` there in the first place? Did they use to be unused?

Reply:
What was the answer to this?